sql/core/src/main/scala/org/apache/spark/sql/execution/joins/HashedRelation.scala

```diff
@@ -324,7 +324,9 @@ private[joins] object UnsafeHashedRelation {
  *
  * see http://java-performance.info/implementing-world-fastest-java-int-to-int-hash-map/
  */
-private[execution] final class LongToUnsafeRowMap(var mm: TaskMemoryManager, capacity: Int)
+private[execution] final class LongToUnsafeRowMap(
+    @transient val mm: TaskMemoryManager,
```

Comment:
It turns out to be an issue of "logical grouping" vs. "physical grouping" of members or objects. There are pros and cons for both. Right now, probably minimizing the impact on existing codes is preferred.  Thanks.